pkg/build/builder/cmd/builder.go

```diff
@@ -92,60 +92,51 @@ func newBuilderConfigFromEnvironment(out io.Writer, needsDocker bool) (*builderC
-					systemContext.SystemRegistriesConfPath = registriesConfPath
-				}
-			}
-			if registriesDirPath, ok := os.LookupEnv("BUILD_REGISTRIES_DIR_PATH"); ok && len(registriesDirPath) > 0 {
-				if _, err := os.Stat(registriesDirPath); err == nil {
-					systemContext.RegistriesDirPath = registriesDirPath
-				}
+		}
+		if registriesDirPath, ok := os.LookupEnv("BUILD_REGISTRIES_DIR_PATH"); ok && len(registriesDirPath) > 0 {
+			if _, err := os.Stat(registriesDirPath); err == nil {
+				systemContext.RegistriesDirPath = registriesDirPath
 			}
-			if signaturePolicyPath, ok := os.LookupEnv("BUILD_SIGNATURE_POLICY_PATH"); ok && len(signaturePolicyPath) > 0 {
-				if _, err := os.Stat(signaturePolicyPath); err == nil {
-					systemContext.SignaturePolicyPath = signaturePolicyPath
-				}
+		}
+		if signaturePolicyPath, ok := os.LookupEnv("BUILD_SIGNATURE_POLICY_PATH"); ok && len(signaturePolicyPath) > 0 {
+			if _, err := os.Stat(signaturePolicyPath); err == nil {
+				systemContext.SignaturePolicyPath = signaturePolicyPath
 			}
+		}
 
-			storeOptions := storage.DefaultStoreOptions
-			if driver, ok := os.LookupEnv("BUILD_STORAGE_DRIVER"); ok {
-				storeOptions.GraphDriverName = driver
-			}
-			if storageConfPath, ok := os.LookupEnv("BUILD_STORAGE_CONF_PATH"); ok && len(storageConfPath) > 0 {
-				if _, err := os.Stat(storageConfPath); err == nil {
-					storage.ReloadConfigurationFile(storageConfPath, &storeOptions)
-				}
+		storeOptions := storage.DefaultStoreOptions
+		if driver, ok := os.LookupEnv("BUILD_STORAGE_DRIVER"); ok {
+			storeOptions.GraphDriverName = driver
+		}
+		if storageConfPath, ok := os.LookupEnv("BUILD_STORAGE_CONF_PATH"); ok && len(storageConfPath) > 0 {
+			if _, err := os.Stat(storageConfPath); err == nil {
+				storage.ReloadConfigurationFile(storageConfPath, &storeOptions)
 			}
+		}
 
-			store, err := storage.GetStore(storeOptions)
-			cfg.store = store
-			if err != nil {
-				return nil, err
-			}
-			cfg.cleanup = func() {
-				if _, err := store.Shutdown(false); err != nil {
-					glog.V(0).Infof("Error shutting down storage: %v", err)
-				}
+		store, err := storage.GetStore(storeOptions)
+		cfg.store = store
+		if err != nil {
+			return nil, err
+		}
+		cfg.cleanup = func() {
+			if _, err := store.Shutdown(false); err != nil {
+				glog.V(0).Infof("Error shutting down storage: %v", err)
 			}
-			istorage.Transport.SetStore(store)
+		}
+		istorage.Transport.SetStore(store)
 
-			dockerClient, err := bld.GetDaemonlessClient(systemContext, store, os.Getenv("BUILD_ISOLATION"))
-			if err != nil {
-				return nil, fmt.Errorf("no daemonless store: %v", err)
-			}
-			cfg.dockerClient = dockerClient
-			cfg.dockerEndpoint = "n/a"
+		dockerClient, err := bld.GetDaemonlessClient(systemContext, store, os.Getenv("BUILD_ISOLATION"))
+		if err != nil {
+			return nil, fmt.Errorf("no daemonless store: %v", err)
 		}
+		cfg.dockerClient = dockerClient
+		cfg.dockerEndpoint = "n/a"
```

Comment:
Any reason why this can't be the empty string?

Reply:
that's a question for @nalind but this PR retained the existing value.

Reply:
IIRC we used to hit an error in the S2I Validator() function (somewhere around https://github.com/openshift/source-to-image/blob/master/pkg/api/validation/validation.go#L23) if we just left it unset.

Reply:
@bparees probably worth a comment, then (and a future techdebt fix for s2i).

Reply:
commented.